src/aFIL.sol

```diff
@@ -1,25 +1,74 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
+import "openzeppelin-contracts/token/ERC20/ERC20.sol";
 import "./LoanAgent.sol";
 
-// is ERC20
-contract aFIL {
-    modifier onlyOracle() {
-        _;
+error InsufficientTokenAvailable(uint availableAmount, uint requestAmount);
+
+contract aFIL is ERC20 {
+    uint public totalLoanAmount;
+    address public immutable oracle;
+
+    constructor(address _oracle) ERC20("arc FIL", "aFIL") {
+        oracle = _oracle;
     }
 
     struct Loan {
         LoanAgent loanAgent;
-        uint loanAmount;
-        uint finishTime;
+        uint totalAmount;
```

Comment:
aFIL contract will only manage aFIL:FIL pool and should not have any kind of relation with LoanAgents or other things.

Loan and loans should be moved to LoanAgentFactory, as it holds power to create and manage them.

Reply:
Only maintain the total loan amount